In [80]:
import numpy as np
import pandas as pd
import warnings
from pandas.errors import SettingWithCopyWarning

In [82]:
warnings.filterwarnings('ignore', category=SettingWithCopyWarning)

In [9]:
latlong = pd.read_csv('district wise centroids.csv')

In [11]:
latlong.head()

State         District   Latitude  Longitude
0  Andaman and Nicobar  Andaman Islands  12.382571  92.822911
1  Andaman and Nicobar  Nicobar Islands   7.835291  93.511601
2       Andhra Pradesh         Adilabad  19.284514  78.813212
3       Andhra Pradesh        Anantapur  14.312066  77.460158
4       Andhra Pradesh         Chittoor  13.331093  78.927639

In [15]:
latlong.shape

(594, 4)

The data is old so we only get 594 districts (as of 2011)

In [18]:
latlong.isnull().sum()

State        0
District     0
Latitude     0
Longitude    0
dtype: int64

No null values 

In [27]:
latlong.duplicated().sum()

0

No duplicate values 

In [34]:
census = pd.read_csv("india-districts-census-2011.csv")

In [38]:
census.shape

(640, 118)

In [40]:
census.head()

District code         State name District name  Population    Male  Female  \
0              1  JAMMU AND KASHMIR       Kupwara      870354  474190  396164   
1              2  JAMMU AND KASHMIR        Badgam      753745  398041  355704   
2              3  JAMMU AND KASHMIR   Leh(Ladakh)      133487   78971   54516   
3              4  JAMMU AND KASHMIR        Kargil      140802   77785   63017   
4              5  JAMMU AND KASHMIR         Punch      476835  251899  224936   

   Literate  Male_Literate  Female_Literate    SC  ...  \
0    439654         282823           156831  1048  ...   
1    335649         207741           127908   368  ...   
2     93770          62834            30936   488  ...   
3     86236          56301            29935    18  ...   
4    261724         163333            98391   556  ...   

   Power_Parity_Rs_90000_150000  Power_Parity_Rs_45000_150000  \
0                            94                           588   
1                           126                           562   
2                            46                           122   
3                            27                           114   
4                            78                           346   

   Power_Parity_Rs_150000_240000  Power_Parity_Rs_240000_330000  \
0                             71                            101   
1                             72                             89   
2                             15                             22   
3                             12                             18   
4                             35                             50   

   Power_Parity_Rs_150000_330000  Power_Parity_Rs_330000_425000  \
0                            172                             74   
1                            161                             96   
2                             37                             20   
3                             30                             19   
4                             85                             59   

   Power_Parity_Rs_425000_545000  Power_Parity_Rs_330000_545000  \
0                             10                             84   
1                             28                            124   
2                             14                             34   
3                              3                             22   
4                              8                             67   

   Power_Parity_Above_Rs_545000  Total_Power_Parity  
0                            15                1119  
1                            18                1066  
2                            17                 242  
3                             7                 214  
4                            12                 629  

[5 rows x 118 columns]

In [53]:
cols = ['Population','District code','District name','Male','Female','Literate','Households_with_Internet','Housholds_with_Electric_Lighting']

In [57]:
census = census[cols]

In [59]:
census

Population  District code              District name    Male  Female  \
0        870354              1                    Kupwara  474190  396164   
1        753745              2                     Badgam  398041  355704   
2        133487              3                Leh(Ladakh)   78971   54516   
3        140802              4                     Kargil   77785   63017   
4        476835              5                      Punch  251899  224936   
..          ...            ...                        ...     ...     ...   
635       41816            636                       Mahe   19143   22673   
636      200222            637                   Karaikal   97809  102413   
637       36842            638                   Nicobars   20727   16115   
638      105597            639  North  AND Middle Andaman   54861   50736   
639      238142            640              South Andaman  127283  110859   

     Literate  Households_with_Internet  Housholds_with_Electric_Lighting  
0      439654                       762                             83071  
1      335649                      1999                             90190  
2       93770                       574                             17250  
3       86236                       235                             15824  
4      261724                       346                             62900  
..        ...                       ...                               ...  
635     36470                       574                              7150  
636    154916                      1868                             47880  
637     25332                        57                              9301  
638     78683                       388                             18641  
639    190266                      2789                             52479  

[640 rows x 8 columns]

In [84]:
census['Sex Ratio'] = round(census['Male'] / census['Female'] * 100)

In [86]:
census

Population  District code              District name    Male  Female  \
0        870354              1                    Kupwara  474190  396164   
1        753745              2                     Badgam  398041  355704   
2        133487              3                Leh(Ladakh)   78971   54516   
3        140802              4                     Kargil   77785   63017   
4        476835              5                      Punch  251899  224936   
..          ...            ...                        ...     ...     ...   
635       41816            636                       Mahe   19143   22673   
636      200222            637                   Karaikal   97809  102413   
637       36842            638                   Nicobars   20727   16115   
638      105597            639  North  AND Middle Andaman   54861   50736   
639      238142            640              South Andaman  127283  110859   

     Literate  Households_with_Internet  Housholds_with_Electric_Lighting  \
0      439654                       762                             83071   
1      335649                      1999                             90190   
2       93770                       574                             17250   
3       86236                       235                             15824   
4      261724                       346                             62900   
..        ...                       ...                               ...   
635     36470                       574                              7150   
636    154916                      1868                             47880   
637     25332                        57                              9301   
638     78683                       388                             18641   
639    190266                      2789                             52479   

     Sex Ratio  
0        120.0  
1        112.0  
2        145.0  
3        123.0  
4        112.0  
..         ...  
635       84.0  
636       96.0  
637      129.0  
638      108.0  
639      115.0  

[640 rows x 9 columns]

In [92]:
census['Literacy Rate'] = round(census['Literate'] / census['Population'] *100)

In [94]:
census

Population  District code              District name    Male  Female  \
0        870354              1                    Kupwara  474190  396164   
1        753745              2                     Badgam  398041  355704   
2        133487              3                Leh(Ladakh)   78971   54516   
3        140802              4                     Kargil   77785   63017   
4        476835              5                      Punch  251899  224936   
..          ...            ...                        ...     ...     ...   
635       41816            636                       Mahe   19143   22673   
636      200222            637                   Karaikal   97809  102413   
637       36842            638                   Nicobars   20727   16115   
638      105597            639  North  AND Middle Andaman   54861   50736   
639      238142            640              South Andaman  127283  110859   

     Literate  Households_with_Internet  Housholds_with_Electric_Lighting  \
0      439654                       762                             83071   
1      335649                      1999                             90190   
2       93770                       574                             17250   
3       86236                       235                             15824   
4      261724                       346                             62900   
..        ...                       ...                               ...   
635     36470                       574                              7150   
636    154916                      1868                             47880   
637     25332                        57                              9301   
638     78683                       388                             18641   
639    190266                      2789                             52479   

     Sex Ratio  Literacy Rate  
0        120.0           51.0  
1        112.0           45.0  
2        145.0           70.0  
3        123.0           61.0  
4        112.0           55.0  
..         ...            ...  
635       84.0           87.0  
636       96.0           77.0  
637      129.0           69.0  
638      108.0           75.0  
639      115.0           80.0  

[640 rows x 10 columns]

In [102]:
final_df = latlong.merge(census , left_on='District' , right_on ='District name')

In [104]:
final_df

State        District   Latitude  Longitude  Population  \
0    Andhra Pradesh        Adilabad  19.284514  78.813212     2741239   
1    Andhra Pradesh       Anantapur  14.312066  77.460158     4081148   
2    Andhra Pradesh        Chittoor  13.331093  78.927639     4174064   
3    Andhra Pradesh   East Godavari  16.782718  82.243207     5154296   
4    Andhra Pradesh          Guntur  15.884926  80.586576     4887813   
..              ...             ...        ...        ...         ...   
512     West Bengal          Maldah  25.080268  88.171917     3988845   
513     West Bengal     Murshidabad  24.259507  88.168169     7103807   
514     West Bengal           Nadia  23.564110  88.582930     5167600   
515     West Bengal        Puruliya  23.254192  86.396853     2930115   
516     West Bengal  Uttar Dinajpur  26.044619  88.235952     3007134   

     District code   District name     Male   Female  Literate  \
0              532        Adilabad  1369597  1371642   1483347   
1              553       Anantapur  2064495  2016653   2310960   
2              554        Chittoor  2090204  2083860   2667878   
3              545   East Godavari  2569688  2584608   3288577   
4              548          Guntur  2440521  2447292   2960441   
..             ...             ...      ...      ...       ...   
512            332          Maldah  2051541  1937304   2086432   
513            333     Murshidabad  3627564  3476243   4055834   
514            336           Nadia  2653768  2513832   3480555   
515            340        Puruliya  1496996  1433119   1624905   
516            330  Uttar Dinajpur  1551066  1456068   1489667   

     Households_with_Internet  Housholds_with_Electric_Lighting  Sex Ratio  \
0                        5512                            566108      100.0   
1                        8606                            897970      102.0   
2                       13763                            944555      100.0   
3                       22411                           1297273       99.0   
4                       19383                           1186028      100.0   
..                        ...                               ...        ...   
512                      5357                            297716      106.0   
513                      7951                            541169      104.0   
514                     12787                            623729      106.0   
515                      3551                            189889      104.0   
516                      2883                            199307      107.0   

     Literacy Rate  
0             54.0  
1             57.0  
2             64.0  
3             64.0  
4             61.0  
..             ...  
512           52.0  
513           57.0  
514           67.0  
515           55.0  
516           50.0  

[517 rows x 14 columns]

In [108]:
final_df = final_df.drop(columns='District name')

In [110]:
final_df.to_csv('india.csv' , index = False)